lib/web/fetch/body.js

```diff
@@ -29,12 +29,17 @@ try {
 const textEncoder = new TextEncoder()
 function noop () {}
 
-const streamRegistry = new FinalizationRegistry((weakRef) => {
-  const stream = weakRef.deref()
-  if (stream && !stream.locked && !isDisturbed(stream) && !isErrored(stream)) {
-    stream.cancel('Response object has been garbage collected').catch(noop)
-  }
-})
+const hasFinalizationRegistry = globalThis.FinalizationRegistry
+let streamRegistry
+
+if (hasFinalizationRegistry) {
```

Comment:
Why? We support only node 20+. So we always have FinalizationRegistry. Is this some kind of artifact from undici 6?

Reply:
I simply reverted https://github.com/nodejs/undici/commit/c5e74641f40ee12b9f0b101235e792f476a62c20 so this is what was in the codebase prior to https://github.com/nodejs/undici/pull/4320

I can make any changes as needed, but thought that it would be easier to review this way